getconf/base.py

```diff
@@ -275,6 +276,34 @@ def getfloat(self, key, default=0.0, doc=''):
             logger.exception("Unable to cast %s as float for the key %s.", repr(value), key)
             raise
 
+    def gettimespan(self, key, default='0d', doc=''):
+        """Retrieve a value as a datetime.timedelta."""
+        assert (
+            default is None or isinstance(default, str)
+        ), 'gettimespan("%s", %s) has an invalid default value type.' % (key, repr(default))
+        value = self._get(key, default=default, doc=doc, type_hint='timedelta')
+        if value is None:
+            return None
+
+        conversion = {'d': 'days', 'h': 'hours', 'm': 'minutes', 's': 'seconds'}
+        value, duration = value[:-1], value[-1:]
+        try:
+            value = float(value)
+            duration = conversion[duration]
+        except ValueError:
+            logger.exception("Unable to cast %s as float for the key %s.", repr(value), key)
+            raise
+        except KeyError:
+            logger.exception(
+                "%s is not a valid duration for the key %s (available values are %s).",
+                repr(duration),
+                key,
+                ','.join(conversion.keys()),
```

Comment:
`', '` so that we have a space between values?

Minor: `available` -> `acceptable`... but maybe it's just me! :)